File: FBuild/FBuild.lua

if Build == nil then
   Build = "Release" 
end

if Build ~= "Release" and Build ~= "Debug" then 
   error("Build='<Release|Debug>' expected");
end;

local dep = {
   Outdir      = Build;  
   Includes    = { "/boost_1_49_0" };
   Files       = FBuild.Glob("*.cpp");
};

local outOfDate = FBuild.CppOutOfDate(dep);

local compileOptions = {
   Config            = Build;                
   Outdir            = Build; 
   Includes          = { "/boost_1_49_0" };
   CRT               = 'Static';
   Files             = outOfDate;
   PrecompiledHeader = {         
      Header = 'Precompiled.h';  
      Cpp    = 'Precompiled.cpp';
   };
};


FBuild.Compile(compileOptions); 


local linkOptions = {
   Config = Build;                
   Output = "../" .. Build .. "/FBuild.exe";
   Libpath = { "/boost_1_49_0/stage/lib", "../" .. Build };
   Libs = { 
      "Lua.lib",
      "Shlwapi.lib",
   };
   Files = FBuild.Glob(Build, "*.obj");
   
};

if FBuild.FileOutOfDate(linkOptions.Output, linkOptions.Files) then
   FBuild.Link(linkOptions); 
end                      

File: FBuild/FBuild.js

if (args.build == undefined) args.build = "Release";
if (args.build != "Debug" && args.build != "Release") throw "build='<Release|Debug>' expected";

if (args.rebuild == undefined) args.rebuild = false;

ToolChain("x64");

var exe = new Exe;
exe.Build(args.build);
exe.Files(Glob("*.cpp"));
exe.DependencyCheck(!args.rebuild);
exe.CRT("Static");
exe.Defines("_CRT_SECURE_NO_WARNINGS");
exe.PrecompiledHeader("Precompiled.h", "Precompiled.cpp");
exe.WarningLevel(4).WarningAsError(true);

exe.Output("../" + args.build + "/FBuild.exe");
exe.LibPath("../" + args.build);
exe.Libs("Duktape.lib", "Ws2_32.lib", "Winmm.lib", "Shlwapi.lib");
exe.LinkArgs("/SUBSYSTEM:CONSOLE /OPT:ICF=4")

exe.Create();
